File: js/leetcode/easy/169-majority-elements.js
function majorityElement(nums) {
  let hash = {};
  for (let i = 0; i < nums.length; i++) {
    hash[nums[i]] = hash[nums[i]] + 1 || 1;
    if (hash[nums[i]] > nums.length / 2) return nums[i];
  }
}

console.log(majorityElement([2, 2, 1, 1, 1, 2, 2]));

File: c#/leetcode/169-majority-elements.cs
public class Solution
{
  public int MajorityElement(int[] nums)
  {
    int n = nums.Length;
    int majorityFreq = n / 2;

    var dict = new Dictionary<int, int>();

    foreach (int num in nums)
    {
      if (!dict.ContainsKey(num))
      {
        dict.Add(num, 1);
      }
      else
      {
        dict[num]++;
      }
    }

    int[] res = dict.Where(x => x.Value > majorityFreq).Select(x => x.Key).ToArray();
    return res[0];
  }
}